src/SSHConfigFile.php

```diff
@@ -104,9 +104,9 @@ public function findConfiguredHost($host)
         list($user, $host) = $this->parseHost($host);
 
         foreach ($this->groups as $group) {
-            if ((isset($group['host']) == $host && $group['host'] == $host) ||
+            if ((isset($group['host']) && $group['host'] == $host) ||
                 (isset($group['hostname']) && $group['hostname'] == $host)) {
-                if (! empty($user) && isset($group['user']) && $group['user'] != $user) {
+                if (! empty($user) && (! isset($group['user']) || (isset($group['user']) && $group['user'] != $user))) {
```

Comment:
Can we extract this entire `if` into a few methods? It's essentially impossible to even tell what's happening.